src/core.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs')
+
+const fsReaddirRecursive = require('fs-readdir-recursive')
+const Markdown = require('markdown-it')
+const meta = require('markdown-it-meta')
+const md = new Markdown().use(meta)
+const readYaml = require('read-yaml')
+
+const defaultSettings = {
+  folders: {
+    assets: './assets',
+    config: './config',
+    pages: './pages',
+    dist: './public',
+    views: './views'
+  }
+}
+
+const appData = {
+  app: readYaml.sync(`${defaultSettings.folders.config}/app.yaml`),
+  content: null,
+  meta: null,
+  pages: fsReaddirRecursive(defaultSettings.folders.pages),
+  plugins: [],
+  pagesData: []
+}
+
+const getPathName = pagePath => {
+  const pathElements = pagePath.split('/')
+  return pathElements.length > 0
+    ? pathElements.splice(0, pathElements.length - 1).join('/') : '/'
+}
+
+const getPagesData = pages => {
+  return pages.map(page => ({
+    content: md.render(fs.readFileSync(`${defaultSettings.folders.pages}/${page}`, 'utf-8')),
+    meta: Object.assign({}, md.meta, {
+      createdAt: fs.statSync(`${defaultSettings.folders.pages}/${page}`).birthtime,
+      htmlPathName: `/${page.split('.md')[0]}.html`,
+      pagePathName: getPathName(page) // path name without file
```

Comment:
maybe the whole `getPathName` function can be replaced with the `path.dirname` function from the `path` module
https://nodejs.org/api/path.html#path_path_dirname_path

Reply:
Thanks. Updated the code.